### ISSUE TITLE
markdown ==== and ---- header fold incorrect with codeblocks bug

### ISSUE THREAD

**Issue by flyxyz123** [neovim/neovim#14858]
# TLDR
this:
https://github.com/neovim/neovim/blob/7da86f55a7ab08fee8479b219b2fbaf2eb424724/runtime/ftplugin/markdown.vim#L22-L24
should change `v:lnum` to `a:lnum`:
```
function! s:NotCodeBlock(lnum) abort
  return synIDattr(synID(a:lnum, 1, 1), 'name') !=# 'markdownCode'
endfunction
```

# MAIN

- `nvim --version`: v0.5.0-dev+1424-g96d83e2a6 on arch linux, v0.5.0-dev+1421-ga72dfec85 on windows 10
- Operating system/version: arch linux and windows 10
- Terminal name/version:alacritty
- `$TERM`:alacritty on arch linux, xterm-256color on windows 10

Vim also have this problem since neovim and vim have same markdown.vim in runtime/ftplugin/.

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC test.md
:let g:markdown_folding=1
:e test.md
za
```
test.md file see attachment:
[test.md](https://github.com/neovim/neovim/files/6680503/test.md)
test.md content:
````
haha
===========

```
-----------
haha
hoho
```

hoho
----------

some

notes

hoho
----------

some

notes
````

### Actual behaviour

![actual_bug1](https://user-images.githubusercontent.com/30398651/122639367-54c4c280-d0ae-11eb-8ff1-dff754dbc136.png)

### Expected behaviour

![expected_bug1](https://user-images.githubusercontent.com/30398651/122639371-58f0e000-d0ae-11eb-9926-9441da985fc8.png)

### Fix

see TLDR, seems like a simple variable scope problem\
I'll make a pull request later.

### Others

I discovered this bug when investigating another bug. I'll make an issue and pull request for that bug too.

**Comment by glacambre** [comment]
Vim suffers from [the same issue](https://github.com/vim/vim/blob/master/runtime/ftplugin/markdown.vim#L22-L24) and so does its [upstream](https://github.com/tpope/vim-markdown/blob/9d87cc19179496db50fb6d902987737c4b522e38/ftplugin/markdown.vim#L30-L32). Thus you should send your PR to tpope's vim-markdown repo, which will then be pulled into Vim and Neovim will then pull the changes from Vim :).

**Comment by flyxyz123** [comment]
> Vim suffers from [the same issue](https://github.com/vim/vim/blob/master/runtime/ftplugin/markdown.vim#L22-L24) and so does its [upstream](https://github.com/tpope/vim-markdown/blob/9d87cc19179496db50fb6d902987737c4b522e38/ftplugin/markdown.vim#L30-L32). Thus you should send your PR to tpope's vim-markdown repo, which will then be pulled into Vim and Neovim will then pull the changes from Vim :).

Sure, I am writing another bug issue related to this bug right now. I'll send both PR to tpope's vim-markdown repo after I finished that issue.

**Comment by dundargoc** [comment]
Closing as the neovim doesn't maintain the runtime files. This should be sent to the maintainer (listed at the top of the syntax file) so that they can take care of this.
